### ISSUE TITLE
Initial suggestions for balancing.

### ISSUE THREAD

**Issue by 63502** [pau101/Wings#2]
Most beautiful and smooth flight animations I have ever seen, well done!

Good flight mechanics, but still needs balancing.
My initial suggestions: 
1. Vastly increase the default durability of wings and make flight consume a little extra hunger.
2. Make players take severe damage from running into walls and things, just like with the elytra.

**Comment by pau101** [comment]
I'm glad you like visuals!

The default durabilities are pretty low as is, I was anticipating reports of being too weak, have you tested how far a pair of angel wings can take you?

The hunger consumption is something that crossed my mind, but never got around to that, I'll work on that.

I did begin to implement that form of damage, but due to technical limitations, I've held it off for the time being, but I'll look at it again.

**Comment by 63502** [comment]
Please pardon my massive wall of text response. I usually like to be thorough with ideas I propose, but I made this issue last second really late at night. So yeeeaaah... kinda wanted to do the whole sleep thing more than write. I covered everything properly this time though! 😅

After some testing, I feel that the current wings are actually way too powerful given their flight, durability, and falling mechanics. Sure they only give a couple minutes of sustained flight each, but that tiny bit is enough to travel an extensive distance with no consequences or material cost(beyond initial creation). From my current experience in survival use, the wings give nearly unlimited mobility surprisingly early in the game, and at no cost for anything other than aesthetics. 

What I was proposing was that wings like the ones you have made should be limited with similar if not identical mechanics to sprinting.
1. If a player loses hunger at a rate close to double that of sprinting, the massive distance they could travel could be reasonably curbed by food limits. In early game you have less food, so you can't travel as far. When mid to late game rolls around, fill your inventory with steak, delighted meals, an RF to food converter, or whatever else your mods offer you and go as far as you like. If a players hunger drops to 6(3 full haunches), they cannot flap or take off any more and are forced to glide to the ground if in flight. 
2. Gliding down while treading air(yes, it really does look like treading air/water) should consume about half as much hunger as sprinting. Yeah, you aren't outright flying, but you're still flapping and eliminating all fall damage(without any cost currently). If a player's hunger drops below 4(2 full haunches, and 2 less than the standard sprinting minimum) not only can they no longer fly, they can no longer glide down to safety either and they plummet to their demise.
3. My suggestion for a durability boost was more to accommodate the above hunger based balancing than anything else. If wings have a much larger durability, or are unbreakable, the mechanics above become much more realistic and manageable, and they fit into a survival curve a little bit better. At very least, wings should be repairable in an anvil with some kind of material other than a mending book.

After some thought, I must amend a previous statement. With wings currently as they are, impact damage alone would provide little to no difference in current balance. Sure players wouldn't be able to land headfirst at high speed anymore, but they can still just stop flying and glide to the ground or a perch of choice. The hunger based balancing should work just fine on its own, even if you do not manage to get around the technical limitations you mentioned above.

**Comment by pau101** [comment]
You can test my changes based on your suggestions before I'll release: [download](https://drive.google.com/file/d/1Ecw_Znqtq6hq-c3MQJ7r3AejJje9XnVm/view)

**Comment by 63502** [comment]
Flight consumption is a beast(in a good way), the durability problem is solved completely, and the flight and slow fall minimum hunger limits are working smoothly. I have yet to properly test it in my survival world, but from the results of my superflat testing, everything seems to do exactly what it needs to.

The config file does need to be deleted in order for the new default wing durability to take effect.

That freefall animation is definitely Willhelm worthy. 😆

**Comment by 63502** [comment]
Further testing results acquired.

I would recommend dropping the minimum hunger limit for slow falling to 2(1 bar) instead of 4(2 bars). The "safety net" glide distance appears to be too short to be effective at the moment.

**Comment by 63502** [comment]
The changes in pre2 instantly solved the safety net problem.

I started proper survival testing, and I immediately noticed a small bug. I seem to be experiencing slow fall hunger drain whenever I fall any distance down, even if I don't fall long enough for the glide to kick in.

I actually think the slow fall hunger drain should be cut back as it is putting a massive strain on my general movements, even while I am not flying.

**Comment by pau101** [comment]
Only when the player falls faster than 0.5 blocks/tick exhaustion accumulates: [wings-1.0.4-pre3-1.12.2.jar](https://drive.google.com/file/d/1gUJgvyIRjcj1ZTw-3heV1Dczmp5IYN1f/view)

**Comment by 63502** [comment]
All problems I can find are now solved.

I can't think of anything else for now, so yeah.

10/10 best flight mod ever. 😀 👍

**Comment by Azvalk** [comment]
Hi ^^

Would it be possible for some of these functions to be configurable ?
I saw that we could configure the durability of each wing pair, 
would it be possible to have a possibility to disable durability ?

Would it be possible to be able to configure the consumption of hunger ?

This mod is great, like 63502 said "Most beautiful and smooth flight animations I have ever seen",
but the specificity of great mods, is that they often allow to be configurable on all their function ^^

Oh, and I'd like to suggest thinking about a dyeing system.
take your time :D

**Comment by pau101** [comment]
@Azvalk Yes, disabling durability is done by setting the value to 0, but I'm actually removing the configurable for durability as Craftweaker can achieve that fine, but I am adding hunger consumption config.

**Comment by Azvalk** [comment]
Craftweaker ? '-'
I do not know..
Why remove a configuration that works and force us to use an additional mod? T.T

**Comment by pau101** [comment]
Crafttweaker is the go-to mod for configuring other mods, me providing the functionality is additional maintenance, and most configuration would be being done in the context of a mod pack where it would already be in used.

**Comment by CplPibald** [comment]
Setting max durability to 0 using crafttweaker results in being unable to fly.  

    <wings:bat_wings>.maxDamage = 0;

Player has a full hunger bar, but can't activate flight mode with the key.  When jumping off a high place, they just float slowly down to the ground.

Using the released version of 1.0.4

**Comment by pau101** [comment]
Ah, of course.

**Comment by pau101** [comment]
[Wings 1.0.6](https://minecraft.curseforge.com/projects/wings/files/2616328) is now out to fix that.
